kore/src/Kore/Strategies/Goal.hs

```diff
@@ -846,16 +846,35 @@ simplify'
     => Lens' goal ClaimPattern
     -> goal
     -> Strategy.TransitionT (AppliedRule goal) m goal
-simplify' lensClaimPattern =
-    Lens.traverseOf (lensClaimPattern . field @"left") $ \config -> do
-        let definedConfig =
-                Pattern.andCondition (mkDefined <$> config)
-                $ from $ makeCeilPredicate_ (Conditional.term config)
-        configs <-
-            simplifyTopConfiguration definedConfig
-            >>= SMT.Evaluator.filterMultiOr
-            & lift
-        Foldable.asum (pure <$> configs)
+simplify' lensClaimPattern goal = do
+    let destination =
+            Lens.view (lensClaimPattern . field @"right") goal
+        leftCondition =
+            Lens.view (lensClaimPattern . field @"left") goal
+            & Pattern.withoutTerm
+        requiresSideCondition =
+            SideCondition.assumeTrueCondition leftCondition
+    simplifiedDestination <-
+        OrPattern.observeAllT
+        $ Logic.scatter destination
+        >>= Pattern.simplify requiresSideCondition
+    let goal' =
+            Lens.set
+            (lensClaimPattern . field @"right")
+            (OrPattern.flatten simplifiedDestination)
+            goal
+    simplifyLeftHandSide goal'
```

Comment:
Shouldn't we simplify the left-hand side first? We want the condition on the left-hand side to be fully simplified before we use it as a `SideCondition`. Also, we might be able to remove some branches, in which case it isn't necessary to simplify the right-hand side.